src/Common/src/Interop/Unix/System.Native/Interop.GetUnixArchitecture.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, SetLastError = true)]
+        internal static extern int GetUnixArchitecture();
+
+        internal enum ProcessorArchitecture
+        {
+            x86,
+            x64,
+            ARM
```

Comment:
@jkotas In the API review it was decided to not add Arm64 right now. Adding Arm64 will require a new API Review, not in the scope of this PR.